Icons: update topic (#7414)
DevExpress/devextreme-documentation: concepts/60 Themes and Styles/30 Icons/35 Customize Icons.md

DevExtreme icons are implemented as an icon font. You can customize icons with CSS styles specific to text content such as `color`, `font-size`, `font-weight`, and `text-align`.

> DevExtreme UI components that generate icons assign the `dx-icon` class to all generated icons. Assign a unique `id` to a DevExtreme component and use it alongside the `.dx-icon` selector to customize all generated icons within a component at once. To customize a specific generated icon, use a CSS selector with a specific icon class such as `.dx-icon-refresh`. The following code snippet customizes both a specific icon and all generated icons within a [Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/) component:

Reviewer: ```suggestion
DevExtreme UI components that generate icons assign the `dx-icon` class to all generated icons. Assign a unique `id` to a DevExtreme component and use it alongside the `.dx-icon` selector to customize all generated icons in a component simultaneously. To customize a specific generated icon, use a CSS selector with a specific icon class, such as `.dx-icon-refresh`. The following code snippet customizes both a specific icon and all generated icons in a [Toolbar](/Documentation/Guide/UI_Components/Toolbar/Getting_Started_with_Toolbar/) component:
```

---
##### jQuery  

    <!-- tab: index.js -->
    $(function() {
        $("#toolbar").dxToolbar({
            items: {
                widget: 'dxButton',
                options: {
                    icon: 'back',
                },
            }, {
                widget: 'dxButton',
                options: {
                    icon: 'refresh',
                },

>             },

Reviewer 2: items should be an array that contains two button objects:

```js
items: [{
    widget: 'dxButton',
    options: {
        icon: 'back',
    },
}, {
    widget: 'dxButton',
    options: {
        icon: 'refresh',
    },
}]
```

        });
    });

    <!-- tab: index.css -->
    .dx-icon {

>         color: red !important; /* Customizes all generated icons. */

Reviewer 2: I'd say that using the `!important` flag is not a good practice to suggest to our customers. This can be achieved by using a more specific selector:
```css
#toolbar .dx-icon {
  color: red;
}
```

    <!-- tab: app.component.html -->
    <dx-toolbar id="toolbar">
        <dxi-item
            widget="dxButton"
            [options]="{icon: 'back'}"
        ></dxi-item>
        <dxi-item
            widget="dxButton"
            [options]="{icon: 'refresh'}"
        ></dxi-item>
    </dx-toolbar>

    <!-- tab: app.component.css -->
    ::ng-deep .dx-icon {
        color: red !important; /* Customizes all generated icons. */
    }

    ::ng-deep #toolbar .dx-icon-refresh {
        font-size: 24px; /* Customizes all "refresh" icons within #toolbar. */
    }

##### Vue

    <!-- tab: App.vue -->
    <template>
        <DxToolbar id="toolbar">
            <DxItem
                widget="dxButton"
                :options="backButtonOptions"
            />
            <DxItem
                widget="dxButton"
                :options="refreshButtonOptions"
            />
        </DxToolbar>
    </template>
    <script>
        // ...
        const backButtonOptions = { icon: 'back' }
        const refreshButtonOptions = { icon: 'refresh' }
    </script>
    <style scoped>
        .dx-icon {
            color: red !important; /* Customizes all generated icons. */

>         }

Reviewer 2: I'd say that using the `!important` flag is not a good practice to suggest to our customers. This can be achieved by using a more specific selector:
```css
#toolbar .dx-icon {
  color: red;
}
```

        #toolbar .dx-icon-refresh {
            font-size: 24px; /* Customizes all "refresh" icons within #toolbar. */
        }
    </style>

##### React

    <!-- tab: App.js -->
    // ...
    function App() {
        const backButtonOptions = { icon: 'back' }
        const refreshButtonOptions = { icon: 'refresh' }

        return (
            <Toolbar id="toolbar">
                <Item
                    widget="dxButton"
                    options={backButtonOptions}
                />
                <Item
                    widget="dxButton"
                    options={refreshButtonOptions}
                />
            </Toolbar>
        );
    }

    export default App;

    <!-- tab: App.css -->
    .dx-icon {
        color: red !important; /* Customizes all generated icons. */

>     }

Reviewer 2: I'd say that using the `!important` flag is not a good practice to suggest to our customers. This can be achieved by using a more specific selector:
```css
#toolbar .dx-icon {
  color: red;
}
```

Author: I agree, I added the `!important` flag because I couldn't get the `color` style to apply to a broad selector like `.dx-icon` without it. I have revised this code snippet and now use a style that has a higher likelihood of applying to `.dx-icon` in a wide variety of use cases: `transform`.

Author: Following further discussions I have updated this example once again to customize all generated icons within a specific component. `.dx-icon` is too broad of a selector for most widely-used selectors to work without `!important` tags.
